laws/src/main/scala/org/http4s/laws/discipline/ArbitraryInstances.scala

```diff
@@ -84,14 +84,26 @@ private[http4s] trait ArbitraryInstances {
 
   val genQuotedString: Gen[String] = oneOf(genQDText, genQuotedPair).map(s => s"""\"$s\"""")
 
-  val genTchar: Gen[Char] = oneOf {
-    Seq('!', '#', '$', '%', '&', '\'', '*', '+', '-', '.', '^', '_', '`', '|', '~') ++
+  private val tchars =
+    Set('!', '#', '$', '%', '&', '\'', '*', '+', '-', '.', '^', '_', '`', '|', '~') ++
       ('0' to '9') ++ ('A' to 'Z') ++ ('a' to 'z')
-  }
+
+  val genTchar: Gen[Char] = oneOf(tchars)
 
   val genToken: Gen[String] =
     nonEmptyListOf(genTchar).map(_.mkString)
 
+  val genNonTchar = frequency(
+    4 -> oneOf(Set(0x00.toChar to 0x7f.toChar: _*) -- tchars),
+    1 -> oneOf(0x100.toChar to Char.MaxValue)
+  )
```

Comment:
Something like this turns out to be a much more interesting generator than extended charsets in a lot of HTTP parsing cases.  We could make some considerable improvements throughout this file.